src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -675,7 +675,7 @@ private bool TakeStartLine(SocketInput input)
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                 }
 
-                var requestUrlPath = pathBegin.GetString(pathEnd);
+                var requestUrlPath = pathBegin.GetUtf8String(pathEnd);
```

Comment:
Internationalized Resource Identifiers (IRIs) https://www.ietf.org/rfc/rfc3987.txt Mapping of IRIs to URIs; says yes? Convert to utf8 then url-encode to ascii - so reverting would be a safe transformation?